src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/UtilityExtension.kt

```diff
@@ -482,6 +513,14 @@ class UtilityExtension : Extension() {
                             return@action
                         }
 
+                        if (threads.get(channel) != null && threads.get(channel)!!.save) {
```

Comment:
Stahp this. You don't need to hit the database twice. Or three times, in this instance - you only need to do one `get`.